pyiron_base/generic/units.py

```diff
@@ -0,0 +1,300 @@
+
+        Note: `quantity` should already be a key of unit_dict
+
+        """
+        for label in labels:
+            if quantity in self.unit_dict.keys():
+                self._quantity_dict[label] = quantity
+            else:
+                raise ValueError("Quantity {} is not defined. Use `add_quantity` to register the unit of this label")
+
+    def __getitem__(self, item):
+        """
+        Getter to return corresponding `pint` unit for a given quantity
+
+        Args:
+            item (str):
+
+        Returns:
+            pint.unit.Unit/pint.quantity.Quantity: The corresponding `pint` unit/quantity
+        """
+        if item in self._unit_dict.keys():
+            return self._unit_dict[item]
+        elif item in self._quantity_dict.keys():
+            return self._unit_dict[self._quantity_dict[item]]
+        else:
+            raise ValueError("Quantity/label '{}' not registered in this unit registry".format(item))
+
+    def get_dtype(self, quantity):
+        """
+        Returns the data type in which the quantity will be stored
+
+        Args:
+            quantity (str): The quantity
+
+        Returns:
+            type: Corresponding data type
+        """
+        if quantity in self._unit_dict.keys():
+            return self._dtype_dict[quantity]
+        elif quantity in self._quantity_dict.keys():
+            return self._dtype_dict[self._quantity_dict[quantity]]
+        else:
+            raise ValueError("Quantity/label '{}' not registered in this unit registry".format(quantity))
+
+
+class UnitConverter:
+    """
+
+    Module to handle conversions between two different unit registries mainly use to convert units between codes and
+    pyiron submodules.
+
+    To instantiate this class, you need two units registries: a base units registry and a code registry:
+
+    >>> import pint
+    >>> pint_registry = pint.UnitRegistry()
+    >>> base_units = PyironUnitRegistry()
+    >>> base_units.add_quantity(quantity="energy", unit=pint_registry.eV)
+    >>> code_units = PyironUnitRegistry()
+    >>> code_units.add_quantity(quantity="energy",
+    >>>                         unit=1e3 * pint_registry.cal / (pint_registry.mol * pint_registry.N_A))
+    >>> unit_converter = UnitConverter(base_units=base_units, code_units=code_units)
+
+    The unit converter instance can then be used to obtain conversion factors between code and base units either as a
+    `pint` quantity:
+
+    >>> print(unit_converter.code_to_base("energy"))
+    0.043364104241800934 electron_volt
+
+    or as a scalar:
+
+    >>> print(unit_converter.code_to_base_value("energy"))
+    0.043364104241800934
+
+    Alternatively, the unit converter can also be used as decorators for functions that return an array scaled into
+    appropriate units:
+
+    >>> @unit_converter(quantity="energy", conversion="code_to_base")
+    >>>    def return_ones():
+    >>>        return np.ones(5)
```

Comment:
```suggestion
    ... def return_ones():
    ...    return np.ones(5)
```

Reply:
Continuation lines are marked with `...` instead of `>>>` because they indicate the lines as one prompt in a python shell.